ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+
+  m_busy = true;
+  emit isBusy();
+
+  for (const CodedValue &codedValue : m_codedValues)
+  {
+    if(codedValue.name() == choice)
+      m_choiceCodeValue = codedValue;
+  }
+
+  m_featureToEdit->attributes()->replaceAttribute(m_updateFieldName, m_choiceCodeValue.code());
+
+  m_message = QString("Updating feature...");
+  emit messageChanged();
+
+  m_featureToEdit->featureTable()->updateFeatureAsync(m_featureToEdit).then(this, [this]()
+  {
+    m_message = QString("Applying edits...");
+    emit messageChanged();
+    m_serviceGeodatabase->applyEditsAsync().then(this, [this](const QList<FeatureTableEditResult*>& featureTableEditResults)
+    {
+      for (FeatureTableEditResult* featureTableEditResult : featureTableEditResults)
+      {
+        const auto results = featureTableEditResult->editResults();
+        for (FeatureEditResult* featureEditResult : results)
+        {
+          if (!featureEditResult->error().isEmpty())
+          {
+            m_message = QString("Apply edits completed with error.");
+            emit messageChanged();
+            return;
+          }
+          else
+          {
+            m_message = QString("Apply edits completed successfully.\n"
+                "Click 'Get State' to check the updated network state.");
+            emit messageChanged();
+          }
+        }
+      }
+
+      m_busy = false;
+      emit isBusy();
+    }).onFailed(this, [this]()
+    {
+      m_message = QString("Apply edits failed.");
+      emit messageChanged();
+
+      m_busy = false;
+      emit isBusy();
+    });
+  });
+
+  m_deviceFeatureLayer->clearSelection();
+  m_lineFeatureLayer->clearSelection();
+
+  m_isUpdateVisible = false;
+  emit isUpdateVisible();
+
+  m_validateBtn = true;
+  emit validateBtn();
+
+  m_clearBtn = false;
+  emit clearBtn();
+}
+
+void ValidateUtilityNetworkTopology::onClear()
+{
+  // Clear the selection.
+  m_deviceFeatureLayer->clearSelection();
+  m_lineFeatureLayer->clearSelection();
+
+  m_isUpdateVisible = false;
+  emit isUpdateVisible();
+
+  m_featureToEdit = nullptr;
+
+  m_message = QString("Selection cleared.");
+  emit messageChanged();
+
+  m_clearBtn = false;
+  emit clearBtn();
+}
+
+void ValidateUtilityNetworkTopology::onGetState()
+{
+  m_utilityNetwork = m_mapView->map()->utilityNetworks()->first();
+
+  if(m_utilityNetwork->definition()->capabilities()->isSupportsNetworkState())
+  {
+    m_busy = true;
+    emit isBusy();
+
+    m_message = QString("Getting utility network state...");
+    emit messageChanged();
+
+    m_utilityNetwork->stateAsync().then([this](const QFuture<Esri::ArcGISRuntime::UtilityNetworkState*>& state)
+    {
+      m_getstate = state.result();
+
+      // Validate if dirty areas or errors exist
+      m_validateBtn = m_getstate->hasDirtyAreas();
+      emit validateBtn();
+
+      // Trace if network topology is enabled
+      m_traceBtn = m_getstate->isNetworkTopologyEnabled();
```

Comment:
Consider the name of `m_getstate`. What is it getting?

Reply:
changed to `m_utilityNetworkstate`